azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebApp.java

```diff
@@ -0,0 +1,302 @@
+                .webappName(getWebAppClient().name()).build();
+        return new WebAppDeploymentSlot(slotEntity, azureAppService);
+    }
+
+    @Override
+    public List<IWebAppDeploymentSlot> deploymentSlots() {
+        return getWebAppClient().deploymentSlots().list().stream()
+                .map(slot -> new WebAppDeploymentSlot(WebAppDeploymentSlotEntity.builder().id(slot.id()).build(), azureAppService))
+                .collect(Collectors.toList());
+    }
+
+    private com.azure.resourcemanager.appservice.models.WebApp getWebAppClient() {
+        return getWebAppClient(false);
+    }
+
+    synchronized com.azure.resourcemanager.appservice.models.WebApp getWebAppClient(boolean force) {
+        if (webAppClient == null || force) {
+            try {
+                webAppClient = StringUtils.isNotEmpty(entity.getId()) ?
+                        azureClient.webApps().getById(entity.getId()) :
+                        azureClient.webApps().getByResourceGroup(entity.getResourceGroup(), entity.getName());
+                entity = AppServiceUtils.getWebAppEntity(webAppClient);
+            } catch (RuntimeException e) {
+                // SDK will throw exception when resource not founded
+                webAppClient = null;
+            }
+        }
+        return webAppClient;
+    }
+
+    @Override
+    public String id() {
+        return getWebAppClient().id();
+    }
+
+    @Override
+    public String name() {
+        return getWebAppClient().name();
+    }
+
+    public class WebAppCreator extends AbstractAppServiceCreator<WebApp> {
+        // todo: Add validation for required parameters
+        @Override
+        public WebApp commit() {
+            final DefinitionStages.Blank blank = WebApp.this.azureClient.webApps().define(getName());
+            final Runtime runtime = getRuntime();
+            final AppServicePlan appServicePlan = AppServiceUtils.getAppServicePlan(getAppServicePlanEntity(), azureClient);
+            final ResourceGroup resourceGroup = WebApp.this.azureClient.resourceGroups().getByName(getResourceGroup());
+            final DefinitionStages.WithCreate withCreate;
+            switch (runtime.getOperatingSystem()) {
+                case LINUX:
+                    withCreate = createLinuxWebApp(blank, resourceGroup, appServicePlan, runtime);
+                    break;
+                case WINDOWS:
+                    withCreate = createWindowsWebApp(blank, resourceGroup, appServicePlan, runtime);
+                    break;
+                case DOCKER:
+                    final DockerConfiguration dockerConfiguration = getDockerConfiguration().get();
+                    withCreate = createDockerWebApp(blank, resourceGroup, appServicePlan, dockerConfiguration);
+                    break;
+                default:
+                    throw new AzureToolkitRuntimeException(String.format(UNSUPPORTED_OPERATING_SYSTEM, runtime.getOperatingSystem()));
+            }
```

Comment:
I prefer to user if else if to replace switch code.

Reply:
IMO switch may be more clear to handle different os in different ways